src/data_info/services/stat.clj

```diff
@@ -27,6 +28,24 @@
   [stat-map]
   (= (:permission stat-map) :own))
 
+(defn- needs-key?
+  [included-keys needed-key?]
+  (if (contains? included-keys needed-key?)
+    true
+    (cond
+      (= needed-key? :permission)
+      (contains? included-keys :share-count)
+
+      (= needed-key? :type)
+      (not (empty? (cset/intersection #{:infoType :content-type :file-count :dir-count} included-keys)))
```

Comment:
these two conditions are cases where the `is-dir?` and `owns?` things above need to know extra information aside from just what's there already.